### PR TITLE
Issue #5: Highlight remaining buildings, markers on initial buildings

### PR DESCRIPTION
**Pull request by JeremiahTee**
#5 Summary of PR:
Loujain/Zohal work:
- More SGW and Loyola buildings
- Building additional information for those added buildings
- Building images for those added buildings

JT's work:
- Fix buildings.json -> added "center location" coordinates for specific  buildings that will have markers with a custom icon (initials of that building) -> see 08c40b5 for change on buildings.json & implementation of markers
- Added custom icons containing initial for specific buildings -> see 5cf45ce for list of all specific buildings with icons
- Implemented observer pattern to change visibility of building initials' markers depending on the map's camera zoom level -> see dd51515

### CODE REVIEW

**Comment by JeremiahTee** [comment]
This PR may contain work from Zohal that was corrected in her PR. This branch was built upon Zohal's additionalInfoMenu branch (prior to her making the PR)

Might have to fix conflicts when merging -> in that case we pick master version over this branch (for the changes that Zohal corrected in her PR)

**Comment by Amanda-Kolopa** [comment]
Neeham and I quickly looked at the files changed and noticed that these shouldn't be merged:

- app/.settings/org.eclipse.buildship.core.prefs 
- app/.project
- app/.classpath 
- .settings/org.eclipse.buildship.core.prefs
- .project
- .idea/misc.xml

Please delete them.
We will continue to review the pull request in the meantime.

**Comment by Neeham** [comment]
> Neeham and I quickly looked at the files changed and noticed that these shouldn't be merged:
> 
> * app/.settings/org.eclipse.buildship.core.prefs
> * app/.project
> * app/.classpath
> * .settings/org.eclipse.buildship.core.prefs
> * .project
> * .idea/misc.xml
> 
> Please delete them.
> We will continue to review the pull request in the meantime.

I have updated the .gitignore file to prevent those listed files from being pushed onto master as multiple individual have used the same branch so it has been following like a chain.

**Comment by Loujain-Al-Nakeeb** [comment]
> > Neeham and I quickly looked at the files changed and noticed that these shouldn't be merged:
> > 
> > * app/.settings/org.eclipse.buildship.core.prefs
> > * app/.project
> > * app/.classpath
> > * .settings/org.eclipse.buildship.core.prefs
> > * .project
> > * .idea/misc.xml
> > 
> > Please delete them.
> > We will continue to review the pull request in the meantime.
> 
> I have updated the .gitignore file to prevent those listed files from being pushed onto master as multiple individual have used the same branch so it has been following like a chain.
> 
> Please fix the merge conflict. On this branch.

The merge conflict has been fixed 😄 !

**Comment by Amanda-Kolopa** [comment]
Overall amazing work! :smile: 

I just have a small concern. 
There are two files (*Observer* and *Subject*) that are written in Java. 
Can we please change these to Kotlin?

**Comment by JeremiahTee** [comment]
> Overall amazing work! 😄
> 
> I just have a small concern.
> There are two files (_Observer_ and _Subject_) that are written in Java.
> Can we please change these to Kotlin?

Thank you for pointing it out! I must've been really tired last night, didn't even realize I wrote Java code instead of Kotlin 😅. Fixed with d31eb7a

File: app/src/main/assets/buildings.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
     "SGW_buildings": [{
             "name": "Henry F. Hall Building",
-            "initial": "H",
```
Comment (JeremiahTee) [inline app/src/main/assets/buildings.json]: I forgot to mention in the commit message:
I removed these initials in the json as they are not needed. The initials are actually various icons from this path
...\SOEN390-CampusGuideMap\app\src\main\res\mipmap\ic_building_h.png

Comment (Neeham) [reply app/src/main/assets/buildings.json]: @JeremiahTee could you please remove them? It was added via @Loujain-Al-Nakeeb merging the master onto this branch to fix merge conflicts 😄

Comment (JeremiahTee) [reply app/src/main/assets/buildings.json]: I removed them in 33a8012 😃

**Comment by Amanda-Kolopa** [comment]
> > Overall amazing work! smile
> > I just have a small concern.
> > There are two files (_Observer_ and _Subject_) that are written in Java.
> > Can we please change these to Kotlin?
> 
> Thank you for pointing it out! I must've been really tired last night, didn't even realize I wrote Java code instead of Kotlin . Fixed with [d31eb7a](https://github.com/RobertBeaudenon/SOEN390-CampusGuideMap/commit/d31eb7abe606c8dff72860370c23e9dd93db4ad3)

Haha no worries! :smile:

File: app/src/main/java/com/droidhats/campuscompass/models/Location.kt [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-import com.google.android.gms.maps.model.LatLng
-import com.google.android.gms.maps.model.Polygon
-import com.google.android.gms.maps.model.PolygonOptions
-
-/*
-* Model for location classes
-* Data relating to locations should be stored in this file
-* */
+import com.droidhats.campuscompass.helpers.Observer
+import com.google.android.gms.maps.model.*
```
Comment (Neeham) [inline app/src/main/java/com/droidhats/campuscompass/models/Location.kt]: Avoid using import com.google.android.gms.maps.model.*

Instead use:
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Polygon
import com.google.android.gms.maps.model.Marker
import com.google.android.gms.maps.model.MarkerOptions
import com.google.android.gms.maps.model.PolygonOptions

Comment (zohalmir) [reply app/src/main/java/com/droidhats/campuscompass/models/Location.kt]: fixed!

**Pull request merged**